lib/mix/test/mix_test.exs

```diff
@@ -153,6 +153,130 @@ defmodule MixTest do
       Application.delete_env(:unknown_app, :foo, persistent: true)
     end
 
+    test ":config_path", %{tmp_dir: tmp_dir} do
+      config_path = Path.join(tmp_dir, "config.exs")
+
+      File.write!(config_path, """
+      import Config
+      config :myapp, :foo, 1
+      """)
+
+      Mix.install(
+        [
+          {:install_test, path: Path.join(tmp_dir, "install_test")}
+        ],
+        config_path: config_path
+      )
+
+      assert Application.fetch_env!(:myapp, :foo) == 1
+    after
+      Application.delete_env(:myapp, :foo)
+    end
+
+    test ":config_path and runtime config", %{tmp_dir: tmp_dir} do
+      config_path = Path.join(tmp_dir, "config.exs")
+
+      File.write!(config_path, """
+      import Config
+      config :myapp, :foo, 1
+      """)
+
+      File.write!(Path.join(tmp_dir, "runtime.exs"), """
+      import Config
+      config :myapp, :bar, 2
+      """)
+
+      Mix.install(
+        [
+          {:install_test, path: Path.join(tmp_dir, "install_test")}
+        ],
+        config_path: config_path
+      )
+
+      assert Application.fetch_env!(:myapp, :foo) == 1
+      assert Application.fetch_env!(:myapp, :bar) == 2
+    after
+      Application.delete_env(:myapp, :foo)
+      Application.delete_env(:myapp, :bar)
+    end
+
+    test ":config_path that does not exist" do
+      assert_raise File.Error, ~r/bad.exs": no such file or directory/, fn ->
+        Mix.install([], config_path: "bad.exs")
+      end
+    end
+
+    defmodule GitApp do
+      def project do
+        [
+          app: :git_app,
+          version: "0.1.0",
+          deps: [
+            {:git_repo, "0.1.0", [git: fixture_path("git_repo")]}
+          ]
+        ]
+      end
+    end
+
+    test ":lockfile with first build", %{tmp_dir: tmp_dir} do
+      Mix.Project.push(GitApp)
+      [_latest_rev, rev | _] = get_git_repo_revs("git_repo")
+      lockfile = Path.join(tmp_dir, "lock")
+      Mix.Dep.Lock.write(lockfile, %{git_repo: {:git, fixture_path("git_repo"), rev, []}})
+      Mix.ProjectStack.pop()
+
+      Mix.install(
+        [
+          {:git_repo, git: fixture_path("git_repo")}
+        ],
+        lockfile: lockfile,
+        verbose: true
+      )
+
+      assert_received {:mix_shell, :info, ["* Getting git_repo " <> _]}
+      assert_received {:mix_shell, :info, ["Mix.install/2 using " <> install_dir]}
```

Comment:
@josevalim we talked about introducing a `:cache_id` option to have a deterministic location of the install dir but that still would not be perfect as I'd have to replicate in the test the rest of the path, the tmp dir and the elixir/otp combination. Grabbing the install dir this way should be reliable enough but lmk otherwise!

We also talked about setting MIX_INSTALL_DIR in the test_helper. I didn't do that because in this test we're already doing that in the `setup` block, we always get a fresh directory (using `@tag :tmp_dir`) which has a small benefit over the global thing in the test_helper that we can have multiple tests using the same dependencies (and thus ending up with the same install dir).